client/src/components/RuleBuilder/RuleTargetComponent.vue

```diff
@@ -3,7 +3,7 @@
         class="rule-link dropdown-item"
         href="javascript:void(0)"
         :class="linkClassName"
-        @click="builder.addNewRule(ruleType)"
+        @click="$emit('add-new-rule', ruleType)"
```

Comment:
See other suggestion. Alternatively adjust casing in `RuleCollectionBuilder`

```suggestion
        @click="$emit('addNewRule', ruleType)"
```